sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,219 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division, absolute_import
+
+import numbers
+import numpy as np
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted, column_or_1d
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int or array-like, shape (n_features_,) (default=2)
+        The number of bins to produce. The intervals for the bins are
+        determined by the minimum and maximum of the input data.
+        Raises ValueError if n_bins < 2.
+
+        If n_bins is an array, and there is a categorical feature at index i,
+        n_bins[i] will be ignored.
+
+    categorical_features : int array-like (default=None)
```

Comment:
In this case, None at least means "none". But in general I disapprove of default=None.

Stating defaults in type specs is against numpydoc convention, but I know it's easier to access.